packages/salesforcedx-vscode-core/src/commands/forceLightningComponentCreate.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { CancellableStatusBar, taskViewService } from '../statuses';
+import {
+  CompositeParametersGatherer,
+  ContinueResponse,
+  DirFileNameSelection,
+  SelectDirPath,
+  SelectFileName,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+const LIGHTNING_CMP_EXTENSION = '.cmp';
+
+class ForceLightningComponentCreateExecutor extends SfdxCommandletExecutor<
+  DirFileNameSelection
+> {
+  public build(data: DirFileNameSelection): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_lightning_component_create_text'))
+      .withArg('force:lightning:component:create')
+      .withFlag('--componentname', data.fileName)
+      .withFlag('--outputdir', data.outputdir)
+      .build();
+  }
+
+  public execute(response: ContinueResponse<DirFileNameSelection>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      if (
+        data != undefined &&
+        data.toString() === '0' &&
+        vscode.workspace.rootPath
+      ) {
+        vscode.workspace
+          .openTextDocument(
+            path.join(
+              vscode.workspace.rootPath,
+              response.data.outputdir,
+              // fileName is also used to create a subdirectory for the component in the aura directory
+              response.data.fileName,
+              response.data.fileName + LIGHTNING_CMP_EXTENSION
+            )
+          )
+          .then(document => vscode.window.showTextDocument(document));
```

Comment:
@Sweetman - Let's alter this later when you have put all your commands in.
I am trying to avoid using .. then() to be more consistent. I just want to use await.
We should lift this up into a promise and then await it.
So

```
const document = await vscode.workspace(...);
vscode.window.showTextDocument(document);
```